modules/core/src/test/java/org/apache/ignite/testframework/junits/common/GridCommonAbstractTest.java

```diff
@@ -2810,4 +2815,35 @@ protected DistributedChangeableProperty<Serializable> computeJobWorkerInterruptT
 
         return timeoutProperty;
     }
+
+    /** Clean work directory, excluding snapshot directory. */
+    protected IgniteEx restartWithCleanPersistence(int nodes, Collection<String> caches) throws Exception {
+        stopAllGrids();
+
+        assertTrue(U.delete(Paths.get(U.defaultWorkDirectory(), "cp").toFile()));
+        assertTrue(U.delete(Paths.get(U.defaultWorkDirectory(), DFLT_MARSHALLER_PATH).toFile()));
+
+        deleteNodesDirs(nodes, DFLT_STORE_DIR, DFLT_BINARY_METADATA_PATH, DFLT_WAL_PATH, DFLT_WAL_ARCHIVE_PATH);
+
+        IgniteEx ign = startGrids(nodes);
+
+        ign.cluster().state(ClusterState.ACTIVE);
+
+        // Destroy caches configured with IgniteConfiguration before restoring snapshot.
+        ign.destroyCaches(caches);
```

Comment:
We can avoid explicitly destroying caches  by moving current test to `IgniteClusterSnapshotRestoreSelfTest` and setting `dfltCacheCfg = null;` before `IgniteEx ign = startGrids(nodes);` It also will make `Collection<String> caches` argument unnecessary.

Reply:
Cool, nice catch.